tests/rptest/services/transform_verifier_service.py

```diff
@@ -0,0 +1,332 @@
+        ])
+
+    def deserialize_status(self, buf: str | bytes | bytearray):
+        return TransformVerifierConsumeStatus.deserialize(buf)
+
+    def is_done(self, status: TransformVerifierConsumeStatus) -> bool:
+        for partition, amt in self.validate.records_acked.items():
+            if status.records_recv.get(partition, 0) < amt:
+                return False
+        return True
+
+
+class TransformVerifierService(Service):
+    """
+    A verifier for transforms.
+
+    General usage looks like the following in a test:
+
+    status = TransformVerifierService.oneshot(
+      context=self.test_context,
+      redpanda=self.redpanda,
+      config=TransformVerifierProduceConfig(...),
+    )
+
+    """
+    EXE = "transform-verifier"
+    LOG_PATH = "/tmp/transform-verifier.log"
+
+    logs = {
+        "transform_verifier_log": {
+            "path": LOG_PATH,
+            "collect_default": True
+        }
+    }
+
+    redpanda: RedpandaService
+    report_port: int
+    config: TransformVerifierConsumeConfig | TransformVerifierProduceConfig
+    _pid: typing.Optional[int]
+
+    @classmethod
+    def oneshot(cls, context: TestContext, redpanda: RedpandaService,
+                config: TransformVerifierConsumeConfig
+                | TransformVerifierProduceConfig):
+        """
+        Common pattern to use the service
+        """
+        service = cls(context, redpanda, config)
+        service.start()
+        try:
+            service.wait()
+            final_status = service.get_status()
+            service.stop()
+            service.free()
+            return final_status
+        except:
+            service.stop()
+            raise
+
+    def __init__(self, context: TestContext, redpanda: RedpandaService,
+                 config: TransformVerifierConsumeConfig
+                 | TransformVerifierProduceConfig):
+        super().__init__(context, num_nodes=1)
+        self.redpanda = redpanda
+
+        # Note: using a port that happens to already be in test environment
+        # firewall rules from other use cases.  If changing this, update
+        # terraform for test clusters.
+        self.remote_port = 8080
+        self.config = config
+        self._pid = None
+
+    def clean_node(self, node):
+        self.redpanda.logger.debug(f"{self.__class__.__name__}.clean_node")
+        node.account.kill_process(self.EXE, clean_shutdown=False)
+        if node.account.exists(self.LOG_PATH):
+            node.account.remove(self.LOG_PATH)
+
+    def start_node(self, node):
+        cmd = " ".join([
+            f"/opt/redpanda-tests/go/transform-verifier/{self.EXE}",
+            self.config.serialize_cmd(),
+            f"--brokers={self.redpanda.brokers()}",
+            f"--port={self.remote_port}",
+        ])
+        cmd = f"nohup {cmd} >> {self.LOG_PATH} 2>&1 & echo $!"
+        self.logger.info(f"start_node[{node.name}]: {cmd}")
+        pid_str = node.account.ssh_output(cmd, timeout_sec=10)
```

Comment:
Seems like you could use `RemoteAccount.ssh_capture` here rather than the explicit `nohup`, but then you'd still have to do something fussy to get the pid so 🤷